Fix race found by TestPeerConnection_Renegotiation_CodecChange

```
==================
WARNING: DATA RACE
Write at 0x00c00038b680 by goroutine 36:
  github.com/pion/webrtc/v3.(*RTPTransceiver).setSendingTrack()
      /home/travis/gopath/src/github.com/pion/webrtc/rtptransceiver.go:99 +0x69
  github.com/pion/webrtc/v3.(*PeerConnection).RemoveTrack()
      /home/travis/gopath/src/github.com/pion/webrtc/peerconnection.go:1487 +0x2d2
  github.com/pion/webrtc/v3.TestPeerConnection_Renegotiation_CodecChange()
      /home/travis/gopath/src/github.com/pion/webrtc/peerconnection_renegotiation_test.go:402 +0xb10
  testing.tRunner()
      /home/travis/.gimme/versions/go1.14.linux.amd64/src/testing/testing.go:992 +0x1eb

Previous read at 0x00c00038b680 by goroutine 107:
  github.com/pion/webrtc/v3.(*PeerConnection).checkNegotiationNeeded()
      /home/travis/gopath/src/github.com/pion/webrtc/peerconnection.go:358 +0x780
  github.com/pion/webrtc/v3.(*PeerConnection).negotiationNeededOp()
      /home/travis/gopath/src/github.com/pion/webrtc/peerconnection.go:302 +0x195
  github.com/pion/webrtc/v3.(*PeerConnection).negotiationNeededOp-fm()
      /home/travis/gopath/src/github.com/pion/webrtc/peerconnection.go:274 +0x41
  github.com/pion/webrtc/v3.(*operations).start()
      /home/travis/gopath/src/github.com/pion/webrtc/operations.go:84 +0xd4

Goroutine 36 (running) created at:
  testing.(*T).Run()
      /home/travis/.gimme/versions/go1.14.linux.amd64/src/testing/testing.go:1043 +0x660
  testing.runTests.func1()
      /home/travis/.gimme/versions/go1.14.linux.amd64/src/testing/testing.go:1285 +0xa6
  testing.tRunner()
      /home/travis/.gimme/versions/go1.14.linux.amd64/src/testing/testing.go:992 +0x1eb
  testing.runTests()
      /home/travis/.gimme/versions/go1.14.linux.amd64/src/testing/testing.go:1283 +0x527
  testing.(*M).Run()
      /home/travis/.gimme/versions/go1.14.linux.amd64/src/testing/testing.go:1200 +0x2ff
  main.main()
      _testmain.go:654 +0x347

Goroutine 107 (running) created at:
  github.com/pion/webrtc/v3.(*operations).Enqueue()
      /home/travis/gopath/src/github.com/pion/webrtc/operations.go:35 +0x1e5
  github.com/pion/webrtc/v3.(*PeerConnection).onNegotiationNeeded()
      /home/travis/gopath/src/github.com/pion/webrtc/peerconnection.go:271 +0x1df
  github.com/pion/webrtc/v3.(*PeerConnection).setDescription()
      /home/travis/gopath/src/github.com/pion/webrtc/peerconnection.go:846 +0x469
  github.com/pion/webrtc/v3.(*PeerConnection).SetRemoteDescription()
      /home/travis/gopath/src/github.com/pion/webrtc/peerconnection.go:923 +0x3de
  github.com/pion/webrtc/v3.signalPair()
      /home/travis/gopath/src/github.com/pion/webrtc/peerconnection_test.go:62 +0x369
  github.com/pion/webrtc/v3.TestPeerConnection_Renegotiation_CodecChange()
      /home/travis/gopath/src/github.com/pion/webrtc/peerconnection_renegotiation_test.go:381 +0x5ce
  testing.tRunner()
      /home/travis/.gimme/versions/go1.14.linux.amd64/src/testing/testing.go:992 +0x1eb
==================
```

There is also this flakey fail case

```
=== RUN   TestNegotiationNeededRemoveTrack
    TestNegotiationNeededRemoveTrack: peerconnection_renegotiation_test.go:866: 
        	Error Trace:	peerconnection_renegotiation_test.go:866
        	            				peerconnection.go:317
        	            				operations.go:84
        	            				asm_amd64.s:1373
        	Error:      	Received unexpected error:
        	            	InvalidStateError: connection closed
        	Test:       	TestNegotiationNeededRemoveTrack
    TestNegotiationNeededRemoveTrack: peerconnection_renegotiation_test.go:869: 
        	Error Trace:	peerconnection_renegotiation_test.go:869
        	            				peerconnection.go:317
        	            				operations.go:84
        	            				asm_amd64.s:1373
        	Error:      	Received unexpected error:
        	            	InvalidStateError: connection closed
        	Test:       	TestNegotiationNeededRemoveTrack
    TestNegotiationNeededRemoveTrack: peerconnection_renegotiation_test.go:872: 
        	Error Trace:	peerconnection_renegotiation_test.go:872
        	            				peerconnection.go:317
        	            				operations.go:84
        	            				asm_amd64.s:1373
        	Error:      	Received unexpected error:
        	            	InvalidStateError: connection closed
        	Test:       	TestNegotiationNeededRemoveTrack
    TestNegotiationNeededRemoveTrack: peerconnection_renegotiation_test.go:875: 
        	Error Trace:	peerconnection_renegotiation_test.go:875
        	            				peerconnection.go:317
        	            				operations.go:84
        	            				asm_amd64.s:1373
        	Error:      	Received unexpected error:
        	            	InvalidStateError: connection closed
        	Test:       	TestNegotiationNeededRemoveTrack
    TestNegotiationNeededRemoveTrack: peerconnection_renegotiation_test.go:878: 
        	Error Trace:	peerconnection_renegotiation_test.go:878
        	            				peerconnection.go:317
        	            				operations.go:84
        	            				asm_amd64.s:1373
        	Error:      	Received unexpected error:
        	            	InvalidStateError: connection closed
        	Test:       	TestNegotiationNeededRemoveTrack
    TestNegotiationNeededRemoveTrack: peerconnection_renegotiation_test.go:881: 
        	Error Trace:	peerconnection_renegotiation_test.go:881
        	            				peerconnection.go:317
        	            				operations.go:84
        	            				asm_amd64.s:1373
        	Error:      	Received unexpected error:
        	            	InvalidStateError: connection closed
        	Test:       	TestNegotiationNeededRemoveTrack
panic: sync: negative WaitGroup counter
```

--- a/peerconnection.go
+++ b/peerconnection.go
@@ -355,7 +355,7 @@ func (pc *PeerConnection) checkNegotiationNeeded() bool {
 			// Step 5.3.1
 			if t.Direction() == RTPTransceiverDirectionSendrecv || t.Direction() == RTPTransceiverDirectionSendonly {
 				descMsid, okMsid := m.Attribute(sdp.AttrKeyMsid)
-				if !okMsid || descMsid != t.Sender().track.Msid() {
+				if !okMsid || descMsid != t.Sender().Track().Msid() {
 					return true
 				}
 			}
@@ -1163,8 +1163,8 @@ func (pc *PeerConnection) startRTPSenders(currentTransceivers []*RTPTransceiver)
 			err := transceiver.Sender().Send(RTPSendParameters{
 				Encodings: RTPEncodingParameters{
 					RTPCodingParameters{
-						SSRC:        transceiver.Sender().track.SSRC(),
-						PayloadType: transceiver.Sender().track.PayloadType(),
+						SSRC:        transceiver.Sender().Track().SSRC(),
+						PayloadType: transceiver.Sender().Track().PayloadType(),
 					},
 				}})
 			if err != nil {

--- a/rtpsender.go
+++ b/rtpsender.go
@@ -82,6 +82,12 @@ func (r *RTPSender) Track() *Track {
 	return r.track
 }
 
+func (r *RTPSender) setTrack(track *Track) {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	r.track = track
+}
+
 // Send Attempts to set the parameters controlling the sending of media.
 func (r *RTPSender) Send(parameters RTPSendParameters) error {
 	r.mu.Lock()

--- a/sdp.go
+++ b/sdp.go
@@ -324,8 +324,8 @@ func addTransceiverSDP(d *sdp.SessionDescription, isPlanB bool, dtlsFingerprints
 	}
 
 	for _, mt := range transceivers {
-		if mt.Sender() != nil && mt.Sender().track != nil {
-			track := mt.Sender().track
+		if mt.Sender() != nil && mt.Sender().Track() != nil {
+			track := mt.Sender().Track()
 			media = media.WithMediaSource(track.SSRC(), track.Label() /* cname */, track.Label() /* streamLabel */, track.ID())
 			if !isPlanB {
 				media = media.WithPropertyAttribute("msid:" + track.Label() + " " + track.ID())

